File: project88/src/features/user/TransactionService.js
import transactionApi from "../../api/TransactionAPI";
import { getTransactionByUserID } from "../../redux/slices/transactionSlice";

const TransactionService = {
    getTransaction: async (page, size, filter) => {
        return await transactionApi.getTransaction(page, size, filter);
    },

    getTransactionByUserID: async (userID, page, size, filter) => {
        return await transactionApi.getTransactionByUserID(userID, page, size, filter);
    }
}

export default TransactionService;

File: banking/src/main/java/com/project88/banking/service/TransactionService.java
package com.project88.banking.service;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.stereotype.Service;

import com.project88.banking.dto.TransactionHistoryDTO;
import com.project88.banking.dto.filter.TransactionFilter;
import com.project88.banking.repository.ITransactionRepository;

@Service
public class TransactionService implements ITransactionService {

    @Autowired
    private ITransactionRepository transactionRepository;

    @Override
    public Page<TransactionHistoryDTO> getTransaction(String username, Pageable pageable, TransactionFilter filter) {
        Page<TransactionHistoryDTO> transactionHistoryDTO = transactionRepository.findByUsername(username,
                filter.getStartDate(), filter.getEndDate(), filter.getName(),
                pageable);
        return transactionHistoryDTO;
    }

    @Override
    public Page<TransactionHistoryDTO> getTransactionByUserID(long userID, Pageable pageable,
            TransactionFilter filter) {
        Page<TransactionHistoryDTO> transactionHistoryDTO = transactionRepository
                .findTransactionByUserID(userID, filter.getStartDate(), filter.getEndDate(), filter.getName(),
                        pageable);
        return transactionHistoryDTO;
    }

}
